src/client.rs

```diff
@@ -28,8 +28,8 @@ pub type BoxStreamSync<'a, T> = Pin<Box<dyn Stream<Item = T> + Send + Sync + 'a>
 /// for the client DSL. `S` is the service type, `C` is the substream source.
 #[derive(Debug)]
 pub struct RpcClient<S, C, SInner = S> {
-    source: C,
-    map: Arc<dyn MapService<S, SInner>>,
+    pub(crate) source: C,
+    pub(crate) map: Arc<dyn MapService<S, SInner>>,
```

Comment:
@Frando one idea of this crate was that you should be able to define new patterns outside the crate. For that this would have to be somehow exposed, right?

Reply:
WDYM by "define new patterns outside the crate"?